src/Storages/StorageFile.cpp

```diff
@@ -1104,7 +1106,11 @@ void StorageFile::setStorageMetadata(CommonArguments args)
     storage_metadata.setConstraints(args.constraints);
     storage_metadata.setComment(args.comment);
     setInMemoryMetadata(storage_metadata);
-    setVirtuals(VirtualColumnUtils::getVirtualsForFileLikeStorage(storage_metadata.getColumns()));
+
+    std::string path_for_virtuals;
+    if (args.getContext()->getSettingsRef().use_hive_partitioning && !paths.empty())
+        path_for_virtuals = paths[0];
+    setVirtuals(VirtualColumnUtils::getVirtualsForFileLikeStorage(storage_metadata.getColumns(), args.getContext(), path_for_virtuals, format_settings.value_or(FormatSettings{})));
```

Comment:
If we will move checking use_hive_partitioning to getVirtualsForFileLikeStorage, we can simplify the code:

```suggestion
setVirtuals(VirtualColumnUtils::getVirtualsForFileLikeStorage(storage_metadata.getColumns(), args.getContext(), paths.empty() ? "" : paths[0], format_settings));
```